src/data/constants.js

```diff
@@ -38,16 +37,9 @@ export const LANG_CONFIG = {
   fallbackLng: "en",
   lowerCaseLng: true,
   detection: {
-    order: ["querystring", "navigator"],
-  },
-  resources: {
-    en: {
-      translation: translationEn,
-    },
-    ja: {
-      translation: translationJa,
-    },
+    order: ["querystring", "localStorage", "navigator"],
```

Comment:
I used localStorage because the server doesn't care about the language setting. If there were compatibility issues before, I can change it to use cookies.